core/src/main/java/ai/timefold/solver/core/impl/constructionheuristic/DefaultConstructionHeuristicPhaseFactory.java

```diff
@@ -44,53 +47,65 @@ public DefaultConstructionHeuristicPhaseFactory(ConstructionHeuristicPhaseConfig
         super(phaseConfig);
     }
 
-    @Override
-    public ConstructionHeuristicPhase<Solution_> buildPhase(int phaseIndex, boolean triggerFirstInitializedSolutionEvent,
-            HeuristicConfigPolicy<Solution_> solverConfigPolicy, BestSolutionRecaller<Solution_> bestSolutionRecaller,
-            Termination<Solution_> solverTermination) {
-        ConstructionHeuristicType constructionHeuristicType_ = Objects.requireNonNullElse(
-                phaseConfig.getConstructionHeuristicType(),
+    protected DefaultConstructionHeuristicPhaseBuilder<Solution_> getBaseBuilder(int phaseIndex,
```

Comment:
I understand using the `DefaultConstructionHeuristicPhaseFactory` class to construct both phases, but I wonder if we can use a different approach. Here are some ideas:

Do we want to allow using `RuinRecreateConstructionHeuristicPhase` as an alternative construction phase to `DefaultConstructionHeuristicPhase`?

If so, wouldn't having a separate config class instead of using `ConstructionHeuristicPhaseConfig` make sense?

My point is the use of separate factory and phase classes, which would keep the changes located only in the respective classes, improving the concern separation.

However, if we want to use R&R as an LS method only, I don't see a reason to change the logic in `DefaultConstructionHeuristicPhaseFactory::buildPhase`.

Also, it seems unnecessary to add `isNested` to `DefaultConstructionHeuristicPhase` if the class `RuinRecreateConstructionHeuristicPhase` overrides the methods `doStep` and `phaseEnded`.

In short, can we maintain the default construction factory and phase classes and extend their behavior by creating separate R&R classes?

Reply:
Yeah, I don't like the current (lack of) separation of concerns either. I'll see what can be done.